### DISCUSSION TITLE
Port 25 - inbound?

### DISCUSSION THREAD

**Discussion by JohnHammell** [postalserver/postal#3105] [Help with using Postal] [answered]
Is it necessary for port 25 inbound to point to Postal?

Or is it enough to just have port 25 outbound available for Postal?

What I'm considering is to have Postal setup only for port 25 outbound but port 25 inbound (for the same IP address) I would point to a different email software that is used for receiving emails.

I just want to make sure I won't be messing anything up with Postal by doing this.

In the pre-requisites doc ( https://docs.postalserver.io/getting-started/prerequisites ) it does state:
> One thing to be aware of is you'll need to ensure that your provider does not block port 25 outbound.

So I'm understanding that to mean port 25 outbound is strictly required, but port 25 inbound is not required (for Postal) at all.

Thanks in advance for any insight regarding this.

**Comment by willpower232** [comment, marked as answer]
Technically you don't need port 25 inbound to send emails however you will need it in order to receive bounce messages and incoming mail. It may also be the case that the services receiving the email you're sending will connect to your port 25 to verify the message was sent from a real email server.

**Reply by bitbay** [reply]
Thanks @willpower232 , this clears up some missing info!
The project is awesome, but the documentation definitely seems like a "bare minimum", leaving a lot of open questions.
I recommend adding a list of open ports needed for each scenario (ie. for "sending", for "receiving", accesing web interface, etc.), maybe in FAQ section - or better yet, over at [configuration#ports-and-bind-addresses](https://docs.postalserver.io/getting-started/configuration#ports-and-bind-addresses) with a map of each required port with defaults and a map to the correct environment values to override, as in

| PORT   | inbound                                                                                                                                                  | outbound         | ENV override               |
|--------|----------------------------------------------------------------------------------------------------------------------------------------------------------|------------------|----------------------------|
| 25     | Receive emails to this server from mailservers, send emails from this server through SMTP type credentials and verifying message from sent to mailserver | send emails      | `SMTP_SERVER_DEFAULT_PORT` |
| 80/443 | Postal web interface (http/https)                                                                                                                        | Calling webhooks | N/A                        |

All these information would be really helpful for hardening production systems, configuring firewall rules.
But this is just a sidenote.

My follow up question is, what about other SMTP ports like `465`, `587`, `2525`? Are there used in Postal?
For example, if i set and configure TLS and `SMTP_SERVER_TLS_ENABLED`, what port would be used for incoming/outgoing requests?

Also, i want to disable the web interface once i configure everything - and enable it only for maintenance works "on demand".

My understanding is that currently it is not possible, since dropping incoming on 80 would also mean that my services trying to use the HTTP API with credentials set up won't be able to reach postal.
I had the idea to leverage `caddy`s hot reloading `Caddyfile`s  and disable `postal.example.com:80` completely, but still, since that seems to be a "catch all reverse proxy" i imagine that it will disable HTTP API access as well.

Thanks!

**Reply by willpower232** [reply]
The docs have definitely suffered through their iterations, PRs are always welcome at https://github.com/postalserver/docs

> My follow up question is, what about other SMTP ports like 465, 587, 2525? Are there used in Postal?

You can use port 2525 but you would either have to listen exclusively on it or use something like iptables to juggle the packets and use both 2525 and 25 simultaneously.

Regarding other ports, I believe the expect to be Implicit TLS (i.e. they don't use STARTTLS) so Postal cannot currently be used with them as it needs STARTTLS.

> since dropping incoming on 80 would also mean that my services trying to use the HTTP API with credentials set up won't be able to reach postal

Correct. Presumably the best thing for you would be to explicitly block anything which didn't start with `/api/` or whatever it is?

**Reply by bitbay** [reply]
> The docs have definitely suffered through their iterations, PRs are always welcome at https://github.com/postalserver/docs

I would love to colaborate in the future, but i lack the knowledge about all the ports and their function. Once my crunch is over, i'll look into revisiting the docs and add my "would have loved to know-s on the first run".

> You can use port 2525 but you would either have to listen exclusively on it or use something like iptables to juggle the packets and use both 2525 and 25 simultaneously.

I don't **explicitly want to** use a different port, i was asking about the workings. From Your response i understand that even with TLS configured, the communication (by default) will happen over port 25?

> Correct. Presumably the best thing for you would be to explicitly block anything which didn't start with /api/ or whatever it is?

That definitely will work! I haven't thought about allowing only requests made to `https://postal.example.com/api/`, but since that's the "public" API endpoint i'll do that! Thanks.

**Reply by willpower232** [reply]
> even with TLS configured, the communication (by default) will happen over port 25?

Postal can only listen to one port on its own and also requires explicit STARTTLS so port 25 is the best for it to receive messages on.

Postal doesn't actually care about which port is being used but as 25 is the most basic part of the standard, that is the easiest to go with. Using iptables or similar to route packets from other ports is kind of a trick but there wouldn't be any evidence of it in Postals eyes.

**Comment by JohnHammell** [comment]
The docs only state port 25 outbound being needed.

Regarding bounce messages, in general (not specifically regarding Postal Server), where do bounce messages originate from? Is it the sending server sending the bounce email after being unable to send the email, after the receiving server refuses to accept it?

Or are bounces sent by a receiving server after that server initially accepts the email, then later rejecting & bouncing it?

The distinction would impact whether or not port 25 inbound is needed. If the sending server sends the bounce message, then Postal (being a sending server) would not require port 25 inbound because it originates that bounce email. But if the recipient server originates/sends the bounce back to the sender, after initially accepting it, then maybe port 25 inbound is needed.

The reason I ask these things is because Postal only sends email. It does not handle my inbound email addresses...

So could I safely direct port 25 to an inbound email inbox server software, maybe something like Mail-in-a-Box, Mailcow, Stalwart, etc.

But it seems unclear if it's safe to do this in the sense of safe / not going to break anything with Postal or with anything working properly with email on my server.

**Reply by willpower232** [reply]
> Or are bounces sent by a receiving server after that server initially accepts the email, then later rejecting & bouncing it?

This one.

If Postal is unable to send the message to the receiving server it will be treated as a soft or hard fail.

> The reason I ask these things is because Postal only sends email. It does not handle my inbound email addresses...

Yeah having your domains MX records point at Postal isn't always the best idea. It might be tidier to set up a subdomain of your domain in Postal rather than the top level domain so you can use your own email address with something that has inboxes etc and use Postal to do whatever else.

> So could I safely direct port 25 to an inbound email inbox server software, maybe something like Mail-in-a-Box, Mailcow, Stalwart, etc.

Technically you could set up Postal on port 2525 and then have your own software on port 25 which deals with incoming mail and forwards anything it doesn't know to Postals port.

Bounces would be delivered on the return path email address so in order to use Postal at its best, some incoming messages need to get back to Postal eventually.

**Reply by JohnHammell** [reply]
Thank you for all the info. Very helpful. Last night I configured port 25 inbound to postal to test. Definitely is required to also point port 25 to Postal for the bounce messages to get to Postal. I didn't initially realize that softfail & hardfail did not include bounces.
